crates/agent/src/publications/specs.rs

```diff
@@ -350,6 +350,238 @@ pub fn validate_transition(
     errors
 }
 
+pub async fn enforce_resource_quotas(
```

Comment:
This seems like it needs to account for tasks being disabled

Reply:
Done!